### ISSUE TITLE
TypeError when calling exposed method

### ISSUE THREAD

**Issue by matthew-hodgins** [tomerfiliba/rpyc#255]
I've recently run into an issue where calling any of the exposed methods results in a stack trace similar to this:

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "worker.py", line 1217, in execute_test_suite
    self.server.root.release(test_suite.id)
  File ".../local/lib/python2.7/site-packages/rpyc/core/netref.py", line 151, in __getattribute__
    return syncreq(self, consts.HANDLE_GETATTR, name)
  File ".../local/lib/python2.7/site-packages/rpyc/core/netref.py", line 72, in syncreq
    return conn.sync_request(handler, oid, *args)
  File ".../local/lib/python2.7/site-packages/rpyc/core/protocol.py", line 519, in sync_request
    self.sync_recv_and_dispatch(timeout, True)
  File ".../local/lib/python2.7/site-packages/rpyc/core/protocol.py", line 417, in sync_recv_and_dispatch
    self._dispatch(data)
  File ".../local/lib/python2.7/site-packages/rpyc/core/protocol.py", line 399, in _dispatch
    msg, seq, args = brine.load(data)
TypeError: 'int' object is not iterable
```

I'm connecting to a ThreadedServer. One of the methods in question is get() (which returns an integer or None). The ThreadedServer gets into a state (not sure how yet) where it raises the above TypeError. So far I've been able to work around the issue by putting every server call in a try/except and remaking the rpyc connection but have had no luck finding the root cause.

Any insight would be appreciated. I'm using rpyc version 3.4.4

**Comment by coldfix** [comment]
Hi, thanks for reporting.

Can you provide a minimal server/client example by any chance?

**Comment by matthew-hodgins** [comment]
I've done some more digging and found the root cause. My system is running a server queue (rpyc) and a worker daemon that queries the queue for data. I made a change recently that caused the rpyc connection to be copied to processes spawned by the worker (so the worker and all processes were using the same connection instance).

I fixed the issue by having each spawned process create its own rpyc connection instead of sharing it. Closing this issue as EBKAC!
